### PR TITLE
Fix 'false' in config is interpreted as true

### PR DESCRIPTION
**Pull request by Lehmann-Fabian**

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy [hunk 1]
```diff
@@ -94,7 +94,7 @@ class K8sConfig implements Map<String,Object> {
     }
 
     boolean getCleanup(boolean defValue=true) {
-        target.cleanup == null ? defValue : target.cleanup as boolean
+        target.cleanup == null ? defValue : Boolean.valueOf( target.cleanup as String )
     }
```
[the remark this answers is not in the document]
Comment (Lehmann-Fabian) [reply modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:98]: `cleanup = 'false'` and `cleanup = 'xyz'` evaluate to true which is also weird 🤣

[the remark this answers is not in the document]
Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:98]: well, likely you are right, but it'd be better to track as separate issue at this point

[the remark this answers is not in the document]
Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:98]: oh wait, I was thinking this was a changed in the K8s node fetch feature, fine here then 👍

[the remark this answers is not in the document]
Comment (Lehmann-Fabian) [reply modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:98]: I see. I have too many open Nextflow PRs. Maybe you will find the time to review the others, also the older ones, so we can either merge or close them.

**Review by pditommaso — APPROVED** [review]

**Comment by pditommaso** [comment]
Released along with version `22.05.0-edge`

**Pull request merged**